internal/vault/secrets/ephemeral/kvv2_secret.go

```diff
@@ -0,0 +1,203 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/base"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/client"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/model"
+	"github.com/hashicorp/vault/api"
+	"strconv"
+)
+
+// Ensure the implementation satisfies the resource.ResourceWithConfigure interface
+var _ ephemeral.EphemeralResource = &KVV2EphemeralSecretResource{}
+
+// NewKVV2EphemeralSecretResource returns the implementation for this resource to be
+// imported by the Terraform Plugin Framework provider
+var NewKVV2EphemeralSecretResource = func() ephemeral.EphemeralResource {
+	return &KVV2EphemeralSecretResource{}
+}
+
+// KVV2EphemeralSecretResource implements the methods that define this resource
+type KVV2EphemeralSecretResource struct {
+	base.EphemeralResourceWithConfigure
+}
+
+// KVV2EphemeralSecretModel describes the Terraform resource data model to match the
+// resource schema.
+type KVV2EphemeralSecretModel struct {
+	// common fields to all ephemeral resources
+	base.BaseModelEphemeral
+
+	// fields specific to this resource
+	Mount          types.String `tfsdk:"mount"`
+	Name           types.String `tfsdk:"name"`
+	Version        types.Int32  `tfsdk:"version"`
+	DataJSON       types.String `tfsdk:"data_json"`
+	Data           types.Map    `tfsdk:"data"`
+	CustomMetadata types.Map    `tfsdk:"custom_metadata"`
+	CreatedTime    types.String `tfsdk:"created_time"`
+	DeletionTime   types.String `tfsdk:"deletion_time"`
+	Destroyed      types.Bool   `tfsdk:"destroyed"`
+}
+
+// KVV2EphemeralSecretAPIModel describes the Vault API data model.
+type KVV2EphemeralSecretAPIModel struct {
+	Data     map[string]interface{} `json:"data" mapstructure:"data"`
+	Metadata Metadata               `json:"metadata" mapstructure:"metadata"`
+}
+
+type Metadata struct {
+	CustomMetadata map[string]interface{} `json:"custom_metadata" mapstructure:"custom_metadata"`
+	CreatedTime    string                 `json:"created_time" mapstructure:"created_time"`
+	DeletionTime   string                 `json:"deletion_time" mapstructure:"deletion_time"`
+	Destroyed      bool                   `json:"destroyed" mapstructure:"destroyed"`
+}
+
+// Schema defines this resource's schema which is the data that is available in
+// the resource's configuration, plan, and state
+//
+// https://developer.hashicorp.com/terraform/plugin/framework/resources#schema-method
+func (r *KVV2EphemeralSecretResource) Schema(_ context.Context, _ ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			consts.FieldMount: schema.StringAttribute{
```

Comment:
It would be nice if we could compute this from the MountID. That would simplify the config slightly. Not sure if that is possible?